hoomd/hpmc/pytest/test_pair_angular_step.py

```diff
@@ -0,0 +1,256 @@
+        "directors"]
+    assert angular_step_potential.patch["A"]["deltas"] == particle_dict[
+        "deltas"]
+
+    # after attaching, setting as dict
+    sim = pair_angular_step_simulation_factory()
+    sim.operations.integrator.pair_potentials = [angular_step_potential]
+    sim.run(0)
+    new_particle_dict = dict(directors=[(0, 1, 0)], deltas=[0.2])
+    angular_step_potential.patch["A"] = new_particle_dict
+    assert angular_step_potential.patch["A"]["directors"] == new_particle_dict[
+        "directors"]
+    assert angular_step_potential.patch["A"]["deltas"] == new_particle_dict[
+        "deltas"]
+
+    # after attaching, change the director value
+    angular_step_potential.patch["A"]["directors"] = [(0, 0, 1.0)]
+    assert angular_step_potential.patch["A"]["directors"] == [(0, 0, 1.0)]
+    assert angular_step_potential.patch["A"]["deltas"] == [0.2]
+
+    # after attaching, change the delta value
+    angular_step_potential.patch["A"]["deltas"] = [0.3]
+    assert angular_step_potential.patch["A"]["directors"] == [(0, 0, 1.0)]
+    assert angular_step_potential.patch["A"]["deltas"] == [0.3]
+
+
+@pytest.mark.cpu
+def test_detach(pair_angular_step_simulation_factory, angular_step_potential):
+    particle_dict = dict(directors=[(1.0, 0, 0), (0, 1.0, 0)],
+                         deltas=[0.1, 0.2])
+    angular_step_potential.patch["A"] = particle_dict
+    sim = pair_angular_step_simulation_factory()
+    sim.operations.integrator.pair_potentials = [angular_step_potential]
+    sim.run(0)
+
+    # detach from simulation
+    sim.operations.integrator.pair_potentials.remove(angular_step_potential)
+
+    assert not angular_step_potential._attached
+    assert not angular_step_potential.isotropic_potential._attached
+
+
+def lj(r, r_cut, epsilon, sigma):
+    """Compute the lj energy."""
+    return 4 * epsilon * ((sigma / r)**12 - (sigma / r)**6)
+
+
+# (patch dict,
+#  theta_0, # orientation of the first particle
+#  theta_1,
+#  distance between particles,
+#  expected energy)
+angular_step_test_parameters = [
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        np.pi,
+        5.0,  # >rcut
+        0.0,
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        0,
+        5.0,  # > rcut
+        0.0,
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        np.pi,
+        2.0,  # < rcut
+        lj(2.0, 4.0, 1, 1),
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        0,
+        2.0,  # < rcut
+        0.0,
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        np.pi - 0.099,
+        5.0,  # < rcut
+        0.0,
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
+        0,
+        np.pi - 0.099,
+        2.0,  # < rcut
+        lj(2.0, 4.0, 1, 1),
+    ),
+    (
+        dict(directors=[(1.0, 0, 0)], deltas=[0.1]),
```

Comment:
Please add some tests with two particle types that have different numbers of patches.
For example: 
```
patches['A'] = dict(directors=[(1.0, 0, 0)], deltas=[0.1])
patches['B'] = dict(directors=[(1.0, 0, 0), (0.0, 1.0, 0.0], deltas=[0.1, 0.2])
```
Then rotate the second particle through several angles and ensure that the energy is as expected.

Reply:
I am not sure if my changes match with what you referred to here. I created 2 test sets and 2 separated energy tests for the A-A only and A-B systems. let me know your suggestions.